ext/RMagick/rmmain.c

```diff
@@ -1734,14 +1734,19 @@ version_constants(void)
 {
     const char *mgk_version;
     VALUE str;
+    VALUE nr;
     char long_version[1000];
+    size_t lib_version;
 
-    mgk_version = GetMagickVersion(NULL);
+    mgk_version = GetMagickVersion(&lib_version);
 
     str = rb_str_new2(mgk_version);
     rb_obj_freeze(str);
     rb_define_const(Module_Magick, "Magick_version", str);
 
+    nr = rb_int_new(lib_version);
```

Comment:
Rather than an integer, I'd probably make this a string. That's more conventional in Ruby for version numbers. We can use `Gem::Version` to compare it where we need to. It'll also be more robust that way, I think, since comparing numbers gets a little weird when there are double digits. For example, depending on how you arrange the numbers, `6.10.0` would either show as less than `6.9.0` or greater than `7.0.0`.

Reply:
Using an integer is actually a more robust way of doing this. `6.10.0` would translate to `0x6A0`  and that is `1696` and `0x700` is `1792`. But I just found a bug in the Windows distribution and I have been writing `6.9.10` as `0x6910` instead of `0x69A` for a while now. But I can place a hack here to resolve that.

Reply:
That falls apart as soon as you hit a version that rolls over the hex digit, though. For example, if we had `6.9.16` that is `0x6910`, or 26896. It's also public facing, so it's going to be confusing for any user looking at the version. And as a library maintainer, `Gem::Version` is more intention revealing to me. With a hex number I have to take an extra beat when I'm reading the code to mentally parse the hex number. It's going to be especially confusing if, say, we have any version collisions. For example, we might have `7.1.11` and `7.11.1` at some point. `Gem::Version` already has all the logic we need to compare versions, and it's bundled with Ruby.

Reply:
6.9.16 will never happen because of that. And the version number of ImageMagick 6 is also not very likely to change because we will only do bug fixing there. But I can share your concern and change this into another type. I wonder if it would be possible to create a `Gem::Version` and define that as a constant instead of a string.

Reply:
Using a string is more conventional in Ruby, And it's also more consistent with our existing version constants in `version.rb`.